ArduCopter/events.cpp

```diff
@@ -34,7 +34,12 @@ void Copter::failsafe_radio_on_event()
             desired_action = Failsafe_Action_SmartRTL_Land;
             break;
         case FS_THR_ENABLED_ALWAYS_LAND:
-            desired_action = Failsafe_Action_Land;
+            //    Do not switch to land in Auto mode, Switch to LAND in Manual mode
+            if (flightmode->in_guided_mode() || control_mode == Mode::Number::AUTO) {
```

Comment:
this needs to also check the fs_options bit, so needs to add:
 && failsafe_option(FailsafeOption::RC_CONTINUE_IF_AUTO)

Reply:
Added, also added this for Guided, as we need the same behavior for Guided. Thanks!